integration/build/build_test.go

```diff
@@ -689,6 +689,71 @@ func TestBuildPlatformInvalid(t *testing.T) {
 	assert.Check(t, is.ErrorType(err, errdefs.IsInvalidParameter))
 }
 
+// Test for #47627
+func TestBuildWorkdirNoCacheMiss(t *testing.T) {
```

Comment:
Really minor nit here, that it can be more convenient to put a full link to the ticket here, and perhaps to describe the original issue (so that it may not be needed at all to refer to github. I guess "technically" this comment should also follow `// TestBuildWorkdirNoCacheMiss ....` GoDoc format (but I don't think we have linters verifying GoDoc on tests)

Not a blocker; just if you still want to adjust before merging.